src/function/scalar/list/list_lambdas.cpp

```diff
@@ -0,0 +1,418 @@
+	types.push_back(child_vector.GetType());
+
+	// skip the list column
+	for (idx_t i = 1; i < args.ColumnCount(); i++) {
+		columns.emplace_back(VectorData());
+		args.data[i].Orrify(count, columns[i - 1]);
+		indexes.push_back(0);
+		sel_vectors.emplace_back(SelectionVector(STANDARD_VECTOR_SIZE));
+		types.push_back(args.data[i].GetType());
+	}
+
+	// get the expression executor
+	ExpressionExecutor expr_executor(*lambda_expr);
+	DataChunk input_chunk;
+	DataChunk lambda_chunk;
+
+	// these are only for the list_filter
+	vector<idx_t> lists_len;
+	idx_t curr_list_len = 0, curr_list_offset = 0, appended_lists_cnt = 0, curr_original_list_len = 0;
+	if (!IS_TRANSFORM) {
+		lists_len.reserve(count);
+	}
+
+	// loop over the child entries and create chunks to be executed by the expression executor
+	idx_t elem_cnt = 0;
+	for (idx_t row_idx = 0; row_idx < count; row_idx++) {
+
+		auto lists_index = lists_data.sel->get_index(row_idx);
+		const auto &list_entry = list_entries[lists_index];
+
+		if (!IS_TRANSFORM) {
+			lists_len.push_back(list_entry.length);
+		}
+
+		// set the result to NULL for this row
+		if (!lists_data.validity.RowIsValid(lists_index)) {
+			result_validity.SetInvalid(row_idx);
+			continue;
+		}
+
+		// set the length and offset of the resulting lists of list_transform
+		if (IS_TRANSFORM) {
+			result_entries[row_idx].offset = list_entry.offset;
+			result_entries[row_idx].length = list_entry.length;
+		}
+
+		// empty list, nothing to execute
+		if (list_entry.length == 0) {
+			continue;
+		}
+
+		// get the data indexes
+		for (idx_t col_idx = 0; col_idx < args.ColumnCount() - 1; col_idx++) {
+			indexes[col_idx] = columns[col_idx].sel->get_index(row_idx);
+		}
+
+		// iterate list elements and create transformed expression columns
+		for (idx_t child_idx = 0; child_idx < list_entry.length; child_idx++) {
+
+			// reached STANDARD_VECTOR_SIZE elements, execute
+			if (elem_cnt == STANDARD_VECTOR_SIZE) {
+
+				elem_cnt = 0;
+			}
+
+			// reached STANDARD_VECTOR_SIZE elements
+			if (elem_cnt == STANDARD_VECTOR_SIZE) {
+
+				ExecuteExpression(types, result_types, elem_cnt, sel, sel_vectors, input_chunk, lambda_chunk,
+				                  child_vector, args, expr_executor);
+				auto &lambda_vector = lambda_chunk.data[0];
+
+				if (IS_TRANSFORM) {
+					AppendTransformedToResult(lambda_vector, elem_cnt, result);
+				} else {
+					AppendFilteredToResult(lambda_vector, result_entries, elem_cnt, result, curr_list_len,
+					                       curr_list_offset, appended_lists_cnt, lists_len, curr_original_list_len,
+					                       input_chunk);
+				}
+				elem_cnt = 0;
+			}
+
+			// to slice the child vector
+			auto source_idx = child_data.sel->get_index(list_entry.offset + child_idx);
+			sel.set_index(elem_cnt, source_idx);
+
+			// for each column, set the index of the selection vector to slice properly
+			for (idx_t col_idx = 0; col_idx < args.ColumnCount() - 1; col_idx++) {
```

Comment:
Is this indexing correct?

Reply:
Yes, because `sel_vectors` and `indexes` do not contain the list column.